### PR TITLE
Separator: Add borderWidth style support

### PR DESCRIPTION
**Pull request by ItsJonQ**
This update adds a control for border width for the **Separator** block. This is achieved using the new styled hook/supports registration system and CSS variables.

Screen Capture on 2020-06-23 at 15-10-44

Above is a GIF demo of the control working.

**Note:** How **Separator** appears to be highly dependant on a theme's CSS. For example, TwentyTwenty uses very custom CSS in order to achieve the separator designs. In order to demo this, I had to adjust a CSS value (via Chrome inspect element) to use the variable.

### CSS Variable

The CSS variable for the separator border is:

```
var(--wp--style--border--width);
```

### Opt-in

To opt into this feature, one must add:

```
add_theme_support( "experimental-custom-border" );
```

### Feedback

cc'ing @mtias 😊 . I'd love for some feedback! Does adding this control under a "Border" panel make sense? Also... I'm unsure about the way `<hr />` are rendered by existing CSS. It seems like most users won't see the effect, unless they make some adjustments in their WordPress theme 🤔 

## How has this been tested?

* Add the `add_theme_support` in PHP
* Run `npm run dev`
* Add a Separator block
* (Maybe adjust some CSS in the browser, depending on your theme)
* Adjust the value of the Border Width

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. 

Hopes to resolve:
https://github.com/WordPress/gutenberg/issues/20758

### CODE REVIEW

File: packages/components/src/unit-control/styles/unit-control-styles.js [hunk 1]
```diff
@@ -54,14 +54,16 @@ export const ValueInput = styled( NumberControl )`
 const unitSizeStyles = ( { size } ) => {
 	const sizes = {
 		default: {
-			top: 1,
 			height: 28,
+			lineHeight: '24px',
```
Comment (ItsJonQ) [inline packages/components/src/unit-control/styles/unit-control-styles.js:58]: Adjustments to label height. Abstracted from:
https://github.com/WordPress/gutenberg/pull/23006

**Comment by paaljoachim** [comment]
I just want to say that this is super helpful to be able to adjust the width (height) of the line or dots used. As the default line might be too thin and having the option to make it bigger helps.

**Comment by mtias** [comment]
@ItsJonQ should this be border-width or just height?

**Comment by paaljoachim** [comment]
It would be more natural to call it height. As in how tall the line is. - Vertical.
As the width affects the left/right area. As in how wide is the line. - Horizontal.

So let's say there was a control named height and another control named width. The names in themselves would then make it clear that they do. Height would make the line taller. Width would make the line broader. Expect further in horizontal directions. 

Another scenario with another control.
Let's say that one adds alignment. Left, center and right. User selects to have the line to the left. With a 50% width, and a 5 pixel tall line.

Bottom line is that the Separator block can then have additional controls affecting where on the page it is located (left/center/right), how wide it is, and how tall it is. Hmm I might just have to add an issue or two or so for this. There is some work on Separator styles going on in https://github.com/WordPress/twentytwentyone/issues/278

**Comment by paaljoachim** [comment]
Btw Jon @ItsJonQ 

Check out this comment by @richtabor 
https://github.com/WordPress/gutenberg/issues/16483#issuecomment-509744343
As in how in the animated example shows a readjusting of the height of the block similar to the Spacer block.

**Comment by ItsJonQ** [comment]
> @ItsJonQ should this be border-width or just height?

@mtias I think "size" may be the clearer (and simpler) term to go width 😅 . The CSS property is "width", but visually, we see "height". Size applies to both 😂

**Comment by ItsJonQ** [comment]
It's been a while since this PR was opened, and it looks like a lot has happened with the theme.json / block library settings.
I think it would be easier to close this PR + branch, and start a new 😊

**Comment by simison** [comment]
"Thickness" as the setting name might work well as well, because "height" could imply [the height of the whole area the block takes](https://github.com/WordPress/gutenberg/issues/25989) (including empty space).

**Comment by draganescu** [comment]
I picked this up and so far adding 

```
		"__experimentalBorder": {
			"color": true,
			"radius": true,
			"style": true,
			"width": true,
			"__experimentalDefaultControls": {
				"color": true,
				"radius": true,
				"style": true,
				"width": true
			}
		}
```

to `block.json` brings this:

https://user-images.githubusercontent.com/107534/169897362-32099292-26b5-470e-bef3-f3fd166f3a9e.mp4

The rendering has to also be updated to account for the borders.

**Comment by youknowriad** [comment]
Doing some repository cleanup: Closing this PR as it probably needs to be redone entirely using block support anyway.
